### PR TITLE
depends: patch libool out of libnatpmp/miniupnpc

### PR DESCRIPTION
**Pull request by theuni**
An alternative to https://github.com/bitcoin/bitcoin/pull/29232

Rather than switching to the CMake builds which [proved problematic](https://github.com/bitcoin/bitcoin/pull/29232#issuecomment-1898513919), do the quick and dirty thing of just patching out libtool. Doesn't seem to introduce any new issues.

This should buy us time to upstream the necessary CMake fixes.

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK - maybe upstream will take these patches as well? I guess we can try and send them, given we'll have to send CMake fixes in either case.

Rebased #21778 on top of this rather than #29232.

File: depends/packages/libnatpmp.mk [hunk 1]
```diff
@@ -11,6 +12,10 @@ define $(package)_set_vars
   $(package)_build_env+=CFLAGS="$($(package)_cflags) $($(package)_cppflags)" AR="$($(package)_ar)"
```
Comment (TheCharlatan) [inline depends/packages/libnatpmp.mk:12]: Can the ` $(package)_build_opts_darwin=LIBTOOL="$($(package)_libtool)"` line above (and same for miniupnpc) be removed now? I removed them and things compiled fine.

Comment (theuni) [reply depends/packages/libnatpmp.mk:12]: Yep, thanks for catching this. Updated and squashed.

**Comment by theuni** [comment]
Removed one more missed instance of `LIBTOOL`.

**Review by sedited — APPROVED** [review]
ACK 5b9d5bf866506b22270598aa2dc1269bc02e38e2

Guix build (x86):

```
8b32a57c3b091605070673f6338768f6ee3d5731a553ae96b426827afaf8b8f1  guix-build-5b9d5bf86650/output/aarch64-linux-gnu/SHA256SUMS.part
02a85cac626984fb8d0ea14b449af6ea916c9a3f9f9e3d68137505f5cddded60  guix-build-5b9d5bf86650/output/aarch64-linux-gnu/bitcoin-5b9d5bf86650-aarch64-linux-gnu-debug.tar.gz
f504943f0cea5bfa512a8677c8d0d1968eabcfc5a17d3fc7b810e098d7303019  guix-build-5b9d5bf86650/output/aarch64-linux-gnu/bitcoin-5b9d5bf86650-aarch64-linux-gnu.tar.gz
f976f70dd216792ae78fdee40513cf600f9b9e324e7a42e86f0098129e15216b  guix-build-5b9d5bf86650/output/arm-linux-gnueabihf/SHA256SUMS.part
b6834f3aedac078ea772b38635b5345887d0fb68fa4b9f2f894ad05e1cb8eb0e  guix-build-5b9d5bf86650/output/arm-linux-gnueabihf/bitcoin-5b9d5bf86650-arm-linux-gnueabihf-debug.tar.gz
76058ae27189a14584fd2e3af1aaf721ac7e583902e140b60a308d05ad89ce57  guix-build-5b9d5bf86650/output/arm-linux-gnueabihf/bitcoin-5b9d5bf86650-arm-linux-gnueabihf.tar.gz
33887448717e85acf33b04f8a1464c83312f6b0bd7271c5852164a58eef97f52  guix-build-5b9d5bf86650/output/arm64-apple-darwin/SHA256SUMS.part
55393a180f4fc0539f0a22a16cb67edc659cae70f04cacad44b549d325a990ce  guix-build-5b9d5bf86650/output/arm64-apple-darwin/bitcoin-5b9d5bf86650-arm64-apple-darwin-unsigned.tar.gz
1d9799023fd4646ea6e4bdb7c8a1babf6f7dc86d26bf60f0ee48af84409625f0  guix-build-5b9d5bf86650/output/arm64-apple-darwin/bitcoin-5b9d5bf86650-arm64-apple-darwin-unsigned.zip
13c38211297c307dc28d270164ddb29b2196deee8441e58161eea009d63423c3  guix-build-5b9d5bf86650/output/arm64-apple-darwin/bitcoin-5b9d5bf86650-arm64-apple-darwin.tar.gz
a747043fa34d49eff945045524941ff685c335087e57782f68e0c548159ff805  guix-build-5b9d5bf86650/output/dist-archive/bitcoin-5b9d5bf86650.tar.gz
1d9f57ee098f04e5ff610618e0d99054fc81e0497ccd3770a1cd683935626436  guix-build-5b9d5bf86650/output/powerpc64-linux-gnu/SHA256SUMS.part
d297ffc3be3e9e51157e2e239bfe594a172c4553c8d5259f8d88829567c50aa4  guix-build-5b9d5bf86650/output/powerpc64-linux-gnu/bitcoin-5b9d5bf86650-powerpc64-linux-gnu-debug.tar.gz
7e4870c2da70ba3d02f0cba01ccd1dd3009cbece98d7e37d227d8436ac6c37d4  guix-build-5b9d5bf86650/output/powerpc64-linux-gnu/bitcoin-5b9d5bf86650-powerpc64-linux-gnu.tar.gz
bca0f46c3c6db185bfda1d49ccbe43c2601280c071b5e079867b20a605a63e92  guix-build-5b9d5bf86650/output/powerpc64le-linux-gnu/SHA256SUMS.part
8ea6fe45e8d26b1aff7b6d7d7faa2714117aab9360804d27fe7cfce3386b4c4a  guix-build-5b9d5bf86650/output/powerpc64le-linux-gnu/bitcoin-5b9d5bf86650-powerpc64le-linux-gnu-debug.tar.gz
1926c505251ab255e03591b2e6d214a76243704fb22f0cdd0805dae2fc61054f  guix-build-5b9d5bf86650/output/powerpc64le-linux-gnu/bitcoin-5b9d5bf86650-powerpc64le-linux-gnu.tar.gz
d20c14c72273274ee1b0e50dbb3d4fee1d1341a1867491f6256a327a1145bcc2  guix-build-5b9d5bf86650/output/riscv64-linux-gnu/SHA256SUMS.part
d78190298c511fbc0826be034bda3e42173ce8a72a42a39fd98f8372d15e299e  guix-build-5b9d5bf86650/output/riscv64-linux-gnu/bitcoin-5b9d5bf86650-riscv64-linux-gnu-debug.tar.gz
f7838a3d1275f364bdd95bbe3ad828a36020d0574dc8b6e9ffa210ad5abc0f88  guix-build-5b9d5bf86650/output/riscv64-linux-gnu/bitcoin-5b9d5bf86650-riscv64-linux-gnu.tar.gz
1967ed6640f36e0973480741e4cef7969b68f50b0b6a0e02b1cbfba63f1c6989  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/SHA256SUMS.part
74a8abf167eb0a74ebc772555c83b65d1001ba594e95d2528df75fe6897e814c  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/bitcoin-5b9d5bf86650-x86_64-apple-darwin-unsigned.tar.gz
db49c29e4cb976310b989fedcfcc04427572675ce302de93eb4dc1a550614296  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/bitcoin-5b9d5bf86650-x86_64-apple-darwin-unsigned.zip
144b7ccc1f24217d854d2fbc708e14f5d9fb676c831214f970ec3d4668e0ee92  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/bitcoin-5b9d5bf86650-x86_64-apple-darwin.tar.gz
1359a3aa82e9b080600817a77c0b4af9aba2174ceea670ca8ac20eef062d36c8  guix-build-5b9d5bf86650/output/x86_64-linux-gnu/SHA256SUMS.part
aa77d50f172daaabecf16571bbd5d043747bbc729637e20249f7b7f997ce9029  guix-build-5b9d5bf86650/output/x86_64-linux-gnu/bitcoin-5b9d5bf86650-x86_64-linux-gnu-debug.tar.gz
2404ec35f1a1025085b912d98139909115c42630458a536b275cb2a714c48cff  guix-build-5b9d5bf86650/output/x86_64-linux-gnu/bitcoin-5b9d5bf86650-x86_64-linux-gnu.tar.gz
2cf7dd0540b3a13779710b2172cde7eb7fa479270fe242b89b96b87ae7bfdfcb  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/SHA256SUMS.part
774e5c3cf7efa7a4e1daba9943898e9f47d3feeeafd2aa3d50e9422526eb1951  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64-debug.zip
1bb3ccd4cdaa22631a4c64d8b07fac204d3495126b187b7f8bccc4b0e46d3277  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64-setup-unsigned.exe
1ed5118d90a17fc7d1bfdf9a08cd7237c456535b1dece3e2e3ee502cf3f0a771  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64-unsigned.tar.gz
8e1096abb82c5fb7963fe554e1848e354cb4754f553b763de19148dd7584ae55  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64.zip
```

**Review by fanquake — APPROVED** [review]
ACK 5b9d5bf866506b22270598aa2dc1269bc02e38e2

Guix Build (aarch64):
```bash
8b32a57c3b091605070673f6338768f6ee3d5731a553ae96b426827afaf8b8f1  guix-build-5b9d5bf86650/output/aarch64-linux-gnu/SHA256SUMS.part
02a85cac626984fb8d0ea14b449af6ea916c9a3f9f9e3d68137505f5cddded60  guix-build-5b9d5bf86650/output/aarch64-linux-gnu/bitcoin-5b9d5bf86650-aarch64-linux-gnu-debug.tar.gz
f504943f0cea5bfa512a8677c8d0d1968eabcfc5a17d3fc7b810e098d7303019  guix-build-5b9d5bf86650/output/aarch64-linux-gnu/bitcoin-5b9d5bf86650-aarch64-linux-gnu.tar.gz
f976f70dd216792ae78fdee40513cf600f9b9e324e7a42e86f0098129e15216b  guix-build-5b9d5bf86650/output/arm-linux-gnueabihf/SHA256SUMS.part
b6834f3aedac078ea772b38635b5345887d0fb68fa4b9f2f894ad05e1cb8eb0e  guix-build-5b9d5bf86650/output/arm-linux-gnueabihf/bitcoin-5b9d5bf86650-arm-linux-gnueabihf-debug.tar.gz
76058ae27189a14584fd2e3af1aaf721ac7e583902e140b60a308d05ad89ce57  guix-build-5b9d5bf86650/output/arm-linux-gnueabihf/bitcoin-5b9d5bf86650-arm-linux-gnueabihf.tar.gz
33887448717e85acf33b04f8a1464c83312f6b0bd7271c5852164a58eef97f52  guix-build-5b9d5bf86650/output/arm64-apple-darwin/SHA256SUMS.part
55393a180f4fc0539f0a22a16cb67edc659cae70f04cacad44b549d325a990ce  guix-build-5b9d5bf86650/output/arm64-apple-darwin/bitcoin-5b9d5bf86650-arm64-apple-darwin-unsigned.tar.gz
1d9799023fd4646ea6e4bdb7c8a1babf6f7dc86d26bf60f0ee48af84409625f0  guix-build-5b9d5bf86650/output/arm64-apple-darwin/bitcoin-5b9d5bf86650-arm64-apple-darwin-unsigned.zip
13c38211297c307dc28d270164ddb29b2196deee8441e58161eea009d63423c3  guix-build-5b9d5bf86650/output/arm64-apple-darwin/bitcoin-5b9d5bf86650-arm64-apple-darwin.tar.gz
a747043fa34d49eff945045524941ff685c335087e57782f68e0c548159ff805  guix-build-5b9d5bf86650/output/dist-archive/bitcoin-5b9d5bf86650.tar.gz
1d9f57ee098f04e5ff610618e0d99054fc81e0497ccd3770a1cd683935626436  guix-build-5b9d5bf86650/output/powerpc64-linux-gnu/SHA256SUMS.part
d297ffc3be3e9e51157e2e239bfe594a172c4553c8d5259f8d88829567c50aa4  guix-build-5b9d5bf86650/output/powerpc64-linux-gnu/bitcoin-5b9d5bf86650-powerpc64-linux-gnu-debug.tar.gz
7e4870c2da70ba3d02f0cba01ccd1dd3009cbece98d7e37d227d8436ac6c37d4  guix-build-5b9d5bf86650/output/powerpc64-linux-gnu/bitcoin-5b9d5bf86650-powerpc64-linux-gnu.tar.gz
bca0f46c3c6db185bfda1d49ccbe43c2601280c071b5e079867b20a605a63e92  guix-build-5b9d5bf86650/output/powerpc64le-linux-gnu/SHA256SUMS.part
8ea6fe45e8d26b1aff7b6d7d7faa2714117aab9360804d27fe7cfce3386b4c4a  guix-build-5b9d5bf86650/output/powerpc64le-linux-gnu/bitcoin-5b9d5bf86650-powerpc64le-linux-gnu-debug.tar.gz
1926c505251ab255e03591b2e6d214a76243704fb22f0cdd0805dae2fc61054f  guix-build-5b9d5bf86650/output/powerpc64le-linux-gnu/bitcoin-5b9d5bf86650-powerpc64le-linux-gnu.tar.gz
d20c14c72273274ee1b0e50dbb3d4fee1d1341a1867491f6256a327a1145bcc2  guix-build-5b9d5bf86650/output/riscv64-linux-gnu/SHA256SUMS.part
d78190298c511fbc0826be034bda3e42173ce8a72a42a39fd98f8372d15e299e  guix-build-5b9d5bf86650/output/riscv64-linux-gnu/bitcoin-5b9d5bf86650-riscv64-linux-gnu-debug.tar.gz
f7838a3d1275f364bdd95bbe3ad828a36020d0574dc8b6e9ffa210ad5abc0f88  guix-build-5b9d5bf86650/output/riscv64-linux-gnu/bitcoin-5b9d5bf86650-riscv64-linux-gnu.tar.gz
1967ed6640f36e0973480741e4cef7969b68f50b0b6a0e02b1cbfba63f1c6989  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/SHA256SUMS.part
74a8abf167eb0a74ebc772555c83b65d1001ba594e95d2528df75fe6897e814c  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/bitcoin-5b9d5bf86650-x86_64-apple-darwin-unsigned.tar.gz
db49c29e4cb976310b989fedcfcc04427572675ce302de93eb4dc1a550614296  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/bitcoin-5b9d5bf86650-x86_64-apple-darwin-unsigned.zip
144b7ccc1f24217d854d2fbc708e14f5d9fb676c831214f970ec3d4668e0ee92  guix-build-5b9d5bf86650/output/x86_64-apple-darwin/bitcoin-5b9d5bf86650-x86_64-apple-darwin.tar.gz
1359a3aa82e9b080600817a77c0b4af9aba2174ceea670ca8ac20eef062d36c8  guix-build-5b9d5bf86650/output/x86_64-linux-gnu/SHA256SUMS.part
aa77d50f172daaabecf16571bbd5d043747bbc729637e20249f7b7f997ce9029  guix-build-5b9d5bf86650/output/x86_64-linux-gnu/bitcoin-5b9d5bf86650-x86_64-linux-gnu-debug.tar.gz
2404ec35f1a1025085b912d98139909115c42630458a536b275cb2a714c48cff  guix-build-5b9d5bf86650/output/x86_64-linux-gnu/bitcoin-5b9d5bf86650-x86_64-linux-gnu.tar.gz
2cf7dd0540b3a13779710b2172cde7eb7fa479270fe242b89b96b87ae7bfdfcb  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/SHA256SUMS.part
774e5c3cf7efa7a4e1daba9943898e9f47d3feeeafd2aa3d50e9422526eb1951  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64-debug.zip
1bb3ccd4cdaa22631a4c64d8b07fac204d3495126b187b7f8bccc4b0e46d3277  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64-setup-unsigned.exe
1ed5118d90a17fc7d1bfdf9a08cd7237c456535b1dece3e2e3ee502cf3f0a771  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64-unsigned.tar.gz
8e1096abb82c5fb7963fe554e1848e354cb4754f553b763de19148dd7584ae55  guix-build-5b9d5bf86650/output/x86_64-w64-mingw32/bitcoin-5b9d5bf86650-win64.zip
```

**Pull request merged**